repos/system_upgrade/common/actors/systemfacts/libraries/systemfacts.py

```diff
@@ -61,12 +61,13 @@ def anyhasprefix(value, prefixes):
 @aslist
 def _get_system_users():
     for p in pwd.getpwall():
-        yield User(
-            name=p.pw_name,
-            uid=p.pw_uid,
-            gid=p.pw_gid,
-            home=p.pw_dir
-        )
+        if all([p.pw_name, p.pw_uid, p.pw_gid, p.pw_dir]):
```

Comment:
This needs a comment. Under what circumstances can it happen, that some of the fields are None/empty?

Reply:
I changed the condition as I found out only ``p.pw_dir`` should be checked. The UID and GID are set to 0 by ``pwd`` when not specified however, 0 is also a valid ID, thus it would cause false negatives. The entries in ``/etc/passwd`` are skipped by ``pwd`` if they don't contain username therefore it will be always defined.